### PR TITLE
[Clang Compatibility] Patch to define the .heap section as allocatable.

### PR DESCRIPTION
**Pull request by Johnn333**
An SDK_mods like approach to solving the issue of needing to change nrfx files when we aren't easily able to update the sub-module. Note: The current patch will also apply when building with GCC, from testing this seems be fine in both cases.

### CODE REVIEW

File: nrfx_mods/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This directory enables overwriting nrfx source files so we can instead use our own. nrfx is a submodule of this repository, but this location should be checked first, if any issues arise from this area to ensure default functionality hasn't been changed.
```
Comment (microbit-carlos) [inline nrfx_mods/README.md:1]: Could we also add a note here that if the nrfx library is updated we should check if the updates should also be applied to the files contain here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -33,7 +33,12 @@ list(APPEND INCLUDE_DIRS
 
 list(APPEND SOURCE_FILES "asm/CortexContextSwitch.s")
 list(APPEND SOURCE_FILES "nrfx/mdk/system_${DEV}.c")
-list(APPEND SOURCE_FILES "nrfx/mdk/gcc_startup_${DEV}.S")
+
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/nrfx_mods/mdk/gcc_startup_${DEV}.S")
+    list(APPEND SOURCE_FILES "nrfx_mods/mdk/gcc_startup_${DEV}.S")
+else()
+    list(APPEND SOURCE_FILES "nrfx/mdk/gcc_startup_${DEV}.S")
+endif()
```
Comment (microbit-carlos) [inline CMakeLists.txt:41]: Ah, good catch, I forgot yesterday that there could be other nrf52xxx targets here.

**Comment by JohnVidler** [comment]
Merging this to keep development rolling.

Note that in general we should avoid using this override behaviour where possible, as the maintenance overheads on us increase with each overriden file. In this case this is on a file that is unlikely to change upstream, so its mostly ok.

**Comment by Johnn333** [comment]
Probably worth documenting somewhere that we cannot update nrfx to a higher version whilst using the current SDK: https://github.com/NordicSemiconductor/nrfx/issues/73. Some work will have to be done here to reach the ideal solution of removing method of applying a patch.

**Pull request merged**